app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
Not only we who have notification problems :)

Well, you can check if this tests fail in your setup. For me #6506
doesn't reproduce, so I need some steps to reproduce it.